### PR TITLE
unix,win,build,test: renamed uv_idle_t to uv_spin_t

### PR DESCRIPTION
**Pull request by insipx**
renamed uv_idle_t to uv_spin_t along with related macros,
functions, and tests as per issue [#1035]

"uv_idle_t" and related functions were misleading since they
did not do what they implied. This commit changes the 'uv_idle_t'  type and replaces the 'idle' keyword of related functions with the 'spin' keyword.
Now staying true to their name, spin handles get their callbacks
called on every loop iteration. This was in contradiction to the
previous name 'idle', since the callback was never actually called when
the loop 'idles'.

Renaming Overview:
renamed type calls of 'uv_idle_t' to 'uv_spin_t', renamed the
associated variable names from names similar to 'idle' to
names like 'spin'. Renamed functions 'uv_idle_init',
'uv_idle_start', 'uv_idle_stop' all to their 'spin'
counterparts (EX: uv_spin_start). Renamed macro calls like
XX(IDLE, idle) to XX(SPIN, spin) and typedefs
ending with 'cb' to 'uv_spin_cb' instead of 'uv_idle_cb'.

Refactored and renamed documentation in src/docs/idle.rst to
reflect the changes I made. Renamed idle.rst to spin.rst. The
warning to do with the name 'uv_idle_t' has been removed, since
the issue is being fixed by these changes.

### CODE REVIEW

**Comment by cjihrig** [comment]
@saghul is this what you had in mind?

File: src/unix/core.c [hunk 1]
```diff
@@ -125,8 +125,8 @@ void uv_close(uv_handle_t* handle, uv_close_cb close_cb) {
     uv__check_close((uv_check_t*)handle);
     break;
 
-  case UV_IDLE:
-    uv__idle_close((uv_idle_t*)handle);
+  case UV_SPIN:
+    uv__spin_close((uv_spin_t *) handle);
```
Comment (cjihrig) [inline src/unix/core.c]: The change to the `*` is unrelated and violates the libuv style. Please change all instances of this back.

**Comment by saghul** [comment]
> is this what you had in mind?

I haven't fully reviewed it, but yeah.

File: src/unix/core.c (hunk 1 above)
Comment (InsidiousMind) [inline src/unix/core.c]: For clarity, do you mean the rename from 'uv_idle_t' or the fact that there is a space before `handle` and `*`?

Comment (saghul) [inline src/unix/core.c]: > fact that there is a space before handle and *

This.

**Comment by cjihrig** [comment]
@InsidiousMind if you're interested in picking this back up, it needs a rebase.

**Comment by InsidiousMind** [comment]
This PR is so old i switched my major to poli sci

...

just kidding, i'll get on it :)

**Comment by cjihrig** [comment]
I'm removing the v2 label for now. It can be added back in later, or always picked up in a future major release.

**Pull request closed without merging**